src/app/features/profile/email-verification/email-verification.page.ts

```diff
@@ -0,0 +1,58 @@
+import { Component } from '@angular/core';
+import { TranslocoService } from '@ngneat/transloco';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { defer, timer } from 'rxjs';
+import { catchError, first, map, switchMap, take, tap } from 'rxjs/operators';
+import { BlockingActionService } from '../../../shared/blocking-action/blocking-action.service';
+import { DiaBackendAuthService } from '../../../shared/dia-backend/auth/dia-backend-auth.service';
+import { ErrorService } from '../../../shared/error/error.service';
+
+@UntilDestroy({ checkProperties: true })
+@Component({
+  selector: 'app-email-verification',
+  templateUrl: './email-verification.page.html',
+  styleUrls: ['./email-verification.page.scss'],
+})
+export class EmailVerificationPage {
+  hasSentEmailVerification = false;
+  secondsRemained = 0;
+  readonly RESEND_INTERVAL: number = 60;
+
+  readonly emailVerified$ = this.diaBackendAuthService.emailVerified$;
+  readonly email$ = this.diaBackendAuthService.email$;
+
+  constructor(
+    private readonly blockingActionService: BlockingActionService,
+    private readonly diaBackendAuthService: DiaBackendAuthService,
+    private readonly errorService: ErrorService,
+    private readonly translocoService: TranslocoService
+  ) {}
+
+  async sendEmailVerification() {
+    const ONE_SECOND = 1000;
+    const action$ = defer(() => this.email$).pipe(
+      first(),
+      switchMap(email =>
+        this.diaBackendAuthService.resendActivationEmail$(email)
+      ),
+      catchError((err: unknown) => this.errorService.toastError$(err)),
+      tap(() => {
+        this.hasSentEmailVerification = true;
+        timer(0, ONE_SECOND)
+          .pipe(
+            take(this.RESEND_INTERVAL),
+            map(i => this.RESEND_INTERVAL - i - 1)
+          )
+          .subscribe(val => {
```

Comment:
Nested subscription in RxJS is an [anti-pattern](https://angular-checklist.io/default/checklist/rxjs/Z1eFwa9) best avoiding. Almost under no circumstances would nested subscription be an ideal implementation when writing RxJS. In other words, if you find yourself doing nested subscription, there is almost always a better way to do it.

Here's a few hint on how to rewrite it to avoid nested subscription:

```
const RESEND_COOLDOWN_TICKS = 60;
const TICK_INTERVAL = 1000;
const countdown$ = timer(0, TICK_INTERVAL).pipe(
  take(RESEND_COOLDOWN_TICKS),
  map(tick => RESEND_COOLDOWN_TICKS - tick - 1),
  tap(cooldown => this.secondsRemained = cooldown),
  finalize(() => this.secondsRemained = 0)
);
```

```
return this.blockingActionService
  .run$(action$, {
      message: this.translocoService.translate('message.pleaseWait'),
  })
    .pipe(
      switchMap(() => countdown$),
      untilDestroyed(this)
    )
    .subscribe();
```

In my opinion `switchMap` is safer here since it guarantees the countdown to be reset every time the action is run even if the button is not disabled.

If the button is disabled during countdown, it is effectively the same as `concatMap` and we could use `concatMap` instead.